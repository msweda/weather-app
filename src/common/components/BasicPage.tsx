import {
  createStyles,
  Theme,
  WithStyles,
  withStyles,
} from '@material-ui/core/styles'
import * as React from 'react'

const BasicPage: React.FC<Props> = ({ classes, children }) => {
  return (
    <div className={classes.basicPage}>
      <div className={classes.leftPageMargin} />
      <div className={classes.pageContent}>{children}</div>
      <div className={classes.rightPageMargin} />
    </div>
  )
}

const styles = (theme: Theme) =>
  createStyles({
    basicPage: {
      flex: 1,
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'stretch',
      backgroundColor: theme.palette.background.paper,
    },
    leftPageMargin: {
      flex: 1,
    },
    pageContent: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'stretch',
      width: 800,
      [theme.breakpoints.down('sm')]: {
        width: 600,
      },
      [theme.breakpoints.down('xs')]: {
        width: 300,
      },
      padding: 20,
    },
    rightPageMargin: {
      flex: 1,
    },
  })

interface Props extends WithStyles<typeof styles> {}

export default withStyles(styles)(BasicPage)
