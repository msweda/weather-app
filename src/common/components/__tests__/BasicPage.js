import { createShallow } from '@material-ui/core/test-utils'
import * as React from 'react'
import BasicPage from '../BasicPage'

describe('<BasicPage />', () => {
  let shallow
  let wrapper

  beforeEach(() => {
    shallow = createShallow({ untilSelector: BasicPage })
    wrapper = shallow(<BasicPage />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render wrapped content', () => {
    const mockContent = <p>Test Page</p>
    wrapper = shallow(<BasicPage>{mockContent}</BasicPage>)
    expect(wrapper.contains(mockContent)).toBeTruthy()
  })
})
