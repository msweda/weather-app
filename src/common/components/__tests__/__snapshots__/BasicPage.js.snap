// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicPage /> should match snapshot 1`] = `
<div
  className="BasicPage-basicPage-1"
>
  <div
    className="BasicPage-leftPageMargin-2"
  />
  <div
    className="BasicPage-pageContent-3"
  />
  <div
    className="BasicPage-rightPageMargin-4"
  />
</div>
`;
